packages/flutter/lib/src/animation/scroll_behavior.dart

```diff
@@ -105,26 +105,25 @@ class UnboundedBehavior extends ExtentScrollBehavior {
 Simulation _createFlingScrollSimulation(double position, double velocity, double minScrollOffset, double maxScrollOffset) {
   double startVelocity = velocity * _kSecondsPerMillisecond;
 
-  // Assume that we're rendering at atleast 15 FPS. Stop when we're
-  // scrolling less than one logical pixel per frame. We're essentially
-  // normalizing by the devicePixelRatio so that the threshold has the
-  // same effect independent of the device's pixel density.
-  double endVelocity = 15.0 * ui.window.devicePixelRatio;
+  // Assume that we're rendering at >= 15 FPS. Stop when we're
+  // scrolling less than one physical pixel per frame, which isn't
+  // visually detectable.
+  double endVelocity = 15.0 / ui.window.devicePixelRatio;
 
   // Similar to endVelocity. Stop scrolling when we're this close to
   // destiniation scroll offset.
-  double endDistance = 0.5 * ui.window.devicePixelRatio;
+  double endDistance = 1.0 / ui.window.devicePixelRatio;
 
   SpringDescription spring = new SpringDescription.withDampingRatio(mass: 1.0, springConstant: 170.0, ratio: 1.1);
   ScrollSimulation simulation =
       new ScrollSimulation(position, startVelocity, minScrollOffset, maxScrollOffset, spring, _kScrollDrag)
-    ..tolerance = new Tolerance(velocity: endVelocity.abs(), distance: endDistance);
+    ..tolerance = new Tolerance(velocity: endVelocity, distance: endDistance);
   return simulation;
 }
 
 Simulation _createSnapScrollSimulation(double startOffset, double endOffset, double velocity) {
   double startVelocity = velocity * _kSecondsPerMillisecond;
-  double endVelocity = 15.0 * ui.window.devicePixelRatio * velocity.sign;
+  double endVelocity = velocity.sign * 15.0 / ui.window.devicePixelRatio;
```

Comment:
This shouldn't be hard-coded.

I think we should move all the tolerances into a class at the top level of this file, something like:

```dart
class Tolerances {
  static double endVelocityTolerance = ...; // logical pixels per second
  static double endDistanceTolerance = ...; // logical pixels
}
```

...that way if people are fiddling with transforms they can easily twiddle the values for all the classes and functions here instead of having to do it one by one (or not being able to easily change it at all).